tests/MongoDB.Driver.Core.Tests/Core/Servers/ServerTests.cs

```diff
@@ -266,6 +267,145 @@ public void Dispose_should_dispose_the_server()
             channel.Should().NotBeNull();
         }
 
+        [Theory]
+        [ParameterAttributeData]
+        public void GetChannel_should_update_topology_and_clear_connection_pool_on_network_error_or_timeout(
+            [Values("timeout", "network")] string errorType,
+            [Values(false, true)] bool async)
+        {
+            var serverId = new ServerId(_clusterId, _endPoint);
+            var connectionId = new ConnectionId(serverId);
+            Exception innerMostException;
+            switch (errorType)
+            {
+                case "timeout":
```

Comment:
consider adding this logic here: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Core.TestHelpers/CoreExceptionHelper.cs#L32

Reply:
I think we're unable to add the logic to that method because that method takes in a "Type" and the type for both exceptions here is `SocketException`.  I could create a helper method that takes in a `SocketError` but all that helper method would do is cast the error enum into an int, which I don't think is worth the extra level of indirection. The other option seems to to create a helper method that takes in a string, but that seems like a clunky API even for a test helper method, and again, I'm not sure it's worth the extra level of indirection.

What do you think?

Reply:
we can add overriding for `public static Exception CreateException(Type exceptionType)` which will take only string argument (and will have the current body). See what you think whether it's a good idea or no. I won't insist here

Reply:
Done!